soc/nxp/imxrt/imxrt118x/linker.ld

```diff
@@ -12,8 +12,8 @@
 MEMORY
      {
 
-#if (DT_REG_SIZE(DT_NODELABEL(sdram0)) > 0) && !IS_CHOSEN_SRAM(sdram0)
```

Comment:
I think the soc files should be flexible and driven by DT & kconfig to give folks flexibility at the application level to use what they want.

Reply:
SDRAM on RT1180 isn't out of box. So I think we don't need to consider SDRAM on RT1180 EVK. Default connection on boards is hyperram, so I change the linker description for hyperram, if user don't use hyperram as default RAM region, hyperram region will still display in linker memory region.

Reply:
Sure, but this is an SOC support file. Since the SOC can support SDRAM, shouldn't we keep the definition in the linker file? It seems like all this will do is add an SDRAM linker region if SDRAM is declared in the board devicetree with a nonzero size. Of course, the linker region could also be handled automatically if we use the `zephyr,memory-region` compatible- so this might not be necessary at all.

Reply:
Agreed. I have added back the sdram region (if exist) description in linker file. This ensures that the SOC folder is more static in the future when customers may need support for SDRAM.